### ISSUE TITLE
tmux: versions later than 3.4 break default shell behaviour on darwin

### ISSUE THREAD

**Issue by x123** [NixOS/nixpkgs#348103]
### Describe the bug
The following commit 80931ba844da56228943f61b655a061f7f8a06a3 pushing tmux to 3.5 appears to cause the users shell to not be respected inside of a tmux session on nix-darwin, always falling back to `/bin/sh` no matter if the `default-shell` option is set correctly or not.

### Steps To Reproduce
1. Use tmux 3.5 or later (commit 80931ba844da56228943f61b655a061f7f8a06a3 or newer) on a nix-darwin system
2. Observe that you are always defaulted to `/bin/sh`, despite having another shell defined as default (both natively via `chsh` or by forcing `default-shell` in tmux)

### Expected behavior
tmux respects and uses the correct user shell as it did in 3.4 and below.

### Additional context
For now I need to revert to tmux 3.4, and I'm using the following overlay:
https://github.com/x123/dotfiles/commit/1808a2e5d3fb960d66317d2232a150f23e3b5db3#diff-4c16758b5c5c0626884ec3692e2130e50cc772d436e95e2e0257fbee75b9cfe9R6

### Notify maintainers

@matthiasbeyer 

### Metadata
```console
$ nix run nixpkgs#nix-info -- -m
 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.0.0, macOS 15.0.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.8`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/fpivx4sjcp2vk4rp9nhliln5cwcp3kc6-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Atemu** [comment]
Is this an upstream bug?

**Comment by matthiasbeyer** [comment]
I like to believe that this is an upstream bug. Is there a way we can confirm that this is not an nix issue?

**Comment by x123** [comment]
Good point, I don't have a way to replicate it ATM because all my machines are nixos or nix-darwin. There is a comment here that I think may be involved in the issue:

"Revert to using /bin/sh for #(), run-shell and if-shell - only popups change to use default-shell." (in https://github.com/tmux/tmux/issues/4162)

I will comment on this issue and see if anyone else notices this behavior.

**Comment by x123** [comment]
According to upstream tmux (https://github.com/tmux/tmux/issues/4162) this appears to be a bug in `tmux-sensible` which is being implemented in my case by the `home-manager` `programs.tmux` directives.

"Note that a fix in 3.5 and 3.5a to correctly set SHELL for run-shell and if-shell reveals a bug in the tmux-sensible script, causing it to set default-shell to /bin/sh. This can be worked around by adding this to .tmux.conf after the tmux-sensible script is invoked:"

There is already a bug report in with home-manager: https://github.com/nix-community/home-manager/issues/5952

Probably possible to close this one now, and wait for the HM fix.

**Comment by charlottia** [comment]
@x123 thanks for documenting this issue and what you tracked down so clearly, stopped me pulling my hair out for too long! :)
